pkg/customer/client_test.go

```diff
@@ -0,0 +1,353 @@
+package customer
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/mercadopago/sdk-go/pkg/config"
+	"github.com/mercadopago/sdk-go/pkg/internal/httpclient"
+)
+
+var (
+	createResponseJSON, _ = os.Open("../../resources/mocks/customer/create_response.json")
+	createResponse, _     = io.ReadAll(createResponseJSON)
+
+	searchResponseJSON, _ = os.Open("../../resources/mocks/customer/search_response.json")
+	searchResponse, _     = io.ReadAll(searchResponseJSON)
+
+	getResponseJSON, _ = os.Open("../../resources/mocks/customer/get_response.json")
+	getResponse, _     = io.ReadAll(getResponseJSON)
+
+	updateResponseJSON, _ = os.Open("../../resources/mocks/customer/update_response.json")
+	updateResponse, _     = io.ReadAll(updateResponseJSON)
+)
+
+func TestCreate(t *testing.T) {
+	type fields struct {
+		config *config.Config
+	}
+	type args struct {
+		ctx     context.Context
+		request Request
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *Response
+		wantErr string
+	}{
+		{
+			name: "should_return_error_when_creating_request",
+			fields: fields{
+				config: nil,
+			},
+			args: args{
+				ctx: nil,
+			},
+			want:    nil,
+			wantErr: "error creating request: net/http: nil Context",
+		},
+		{
+			name: "should_return_response",
+			fields: fields{
+				config: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							stringReader := strings.NewReader(string(createResponse))
+							stringReadCloser := io.NopCloser(stringReader)
+							return &http.Response{
+								Body: stringReadCloser,
+							}, nil
+						},
+					},
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+			},
+			want: &Response{
+				ID:       "1681303407-9teKW29W0AXP8X",
+				Email:    "test_user_30851371@testuser.com",
+				UserID:   123456789,
+				LiveMode: false,
+			},
+			wantErr: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &client{
+				cfg: tt.fields.config,
+			}
+			got, err := c.Create(tt.args.ctx, tt.args.request)
+			gotErr := ""
+			if err != nil {
+				gotErr = err.Error()
+			}
+
+			if gotErr != tt.wantErr {
+				t.Errorf("Create() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
para seguir o padrão dos demais clients
```suggestion
				t.Errorf("client.Create() error = %v, wantErr %v", err, tt.wantErr)
```